### PR TITLE
Fix CustomSelectControl component dropdown arrow spacing

### PR DESCRIPTION
**Pull request by getdave**
This PR fixes an issue identified in https://github.com/WordPress/gutenberg/pull/18869#discussion_r435126329 whereby if the selected option in a  `CustomSelectControl` component is a particularly long string then if overlays the down arrow icon.

Screen Shot 2020-06-05 at 08 08 18

This PR adds sufficient padding to the RHS of the control to ensure the icon has room to display (see fix below).

Screen Shot 2020-06-05 at 08 12 47

## How has this been tested?
* Added Storybook story.
* Visually verified issue fixed.

#### On Master
* Go to [latest Storybook for `CustomSelectControl`.](https://wordpress.github.io/gutenberg/?path=/story/components-customselectcontrol--default).
* Select an option from the control dropdown.
* Using devtools amend the select option text to a really long string (see screencapture below for instructions on how to do this).
* See dropdown arrow is overlayed by the text.

Screen Capture on 2020-06-05 at 08-22-16

#### On this PR

* Checkout PR and build Gutenberg `npm run dev`.
* Run local Storybook - `npm run storybook:dev`.
* Navigate to `CustomSelectControl` stories in Storybook. Chose the "Long labels" story.
* See that despite using really long option labels the arrow is _**not**_ overlayed by the option text.

## Types of changes
Bug fix (non-breaking change which fixes an issue).

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/custom-select-control/style.scss [hunk 1]
```diff
@@ -1,3 +1,5 @@
+$icon_width: 24px;
+
```
Comment (talldan) [inline packages/components/src/custom-select-control/style.scss:2]: This is defined already as `$icon-size` in the variables.scss file:
```suggestion
```

File: packages/components/src/custom-select-control/style.scss [hunk 2]
```diff
@@ -16,6 +18,12 @@
 	position: relative;
 	text-align: left;
 
+	// For all button sizes allow sufficient space for the
+	// dropdown "arrow" icon to display.
+	&.components-custom-select-control__button {
+		padding-right: $icon_width;
```
Comment (talldan) [inline packages/components/src/custom-select-control/style.scss:24]:
```suggestion
		padding-right: $icon-size;
```

**Review by talldan — APPROVED** [review]
Looks good, thanks for working this into a separate PR. 

Let's update to use the $icon-size variable then merge.

**Review by marekhrabe — APPROVED** [review]
Notes seem addressed now

**Pull request merged**